### PR TITLE
Add shift filters to csp decorators

### PR DESCRIPTION
**Pull request by riga**
Our CSP (calibrator, selector, producer) decorators already have flags like `data_only` or `mc_only` which control whether the underlying task array function should be applied in case they are evaluated in a dataset that is not data or mc.

Along the same lines, this PR adds additional flags to make a similar decision based on the currently processes shift. It is possible to set

- `nominal_only`, to skip the CSP in case a systematic variation is processed, or
- `shifts_only`, for a more fine-grained control of the shifts that should actually be not skipped.

The implementation is identical through the three decorators.

### CODE REVIEW

**Review by dsavoiu — CHANGES_REQUESTED** [review]
This looks good overall. I only have a comment and a suggestion below (only added for the `Calibrator` class, but they are of course valid also for `Producer` and `Selector`).

File: columnflow/calibration/__init__.py [hunk 1]
```diff
@@ -48,18 +68,30 @@ def decorator(func: Callable) -> DerivableMeta:
             # optionally add skip function
             if mc_only and data_only:
                 raise Exception(f"calibrator {cls_name} received both mc_only and data_only")
-            if mc_only or data_only:
+            if mc_only or data_only or nominal_only:
```
[the remark this answers is not in the document]
Comment (riga) [reply columnflow/calibration/__init__.py:71]: I added this to all CSPs :+1:

File: columnflow/calibration/__init__.py [hunk 2]
```diff
@@ -48,18 +68,30 @@ def decorator(func: Callable) -> DerivableMeta:
             # optionally add skip function
             if mc_only and data_only:
```
[the remark this answers is not in the document]
Comment (riga) [reply columnflow/calibration/__init__.py:69]: Yep :+1: Added.

**Comment by riga** [comment]
Ready to re-review or merge from my side :)

**Review by dsavoiu — APPROVED** [review]
Thanks for the changes! Good to merge from my side.

**Pull request merged**